openedx/core/djangoapps/user_api/accounts/image_helpers.py

```diff
@@ -46,15 +46,18 @@ def _get_profile_image_filename(name, size, file_extension=PROFILE_IMAGE_FILE_EX
     return '{name}_{size}.{file_extension}'.format(name=name, size=size, file_extension=file_extension)
 
 
-def _get_profile_image_urls(name, storage, file_extension=PROFILE_IMAGE_FILE_EXTENSION):
+def _get_profile_image_urls(name, storage, file_extension=PROFILE_IMAGE_FILE_EXTENSION, version=None):
     """
     Returns a dict containing the urls for a complete set of profile images,
     keyed by "friendly" name (e.g. "full", "large", "medium", "small").
     """
-    return {
-        size_display_name: storage.url(_get_profile_image_filename(name, size, file_extension=file_extension))
-        for size_display_name, size in PROFILE_IMAGE_SIZES_MAP.items()
-    }
+    def _make_url(size):  # pylint: disable=missing-docstring
+        url = storage.url(
+            _get_profile_image_filename(name, size, file_extension=file_extension)
+        )
+        return url + '?v=' + version if version is not None else url
```

Comment:
`version=None` is only used for the default profile images.

On Tue, Mar 31, 2015 at 5:45 PM, Max Rothman <notifications@github.com>
wrote:

> In openedx/core/djangoapps/user_api/accounts/image_helpers.py
> <https://github.com/edx/edx-platform/pull/7517#discussion_r27526512>:
>
> >      """
> >      Returns a dict containing the urls for a complete set of profile images,
> >      keyed by "friendly" name (e.g. "full", "large", "medium", "small").
> >      """
> > -    return {
> > -        size_display_name: storage.url(_get_profile_image_filename(name, size, file_extension=file_extension))
> > -        for size_display_name, size in PROFILE_IMAGE_SIZES_MAP.items()
> > -    }
> > +    def _make_url(size):  # pylint: disable=missing-docstring
> > +        url = storage.url(
> > +            _get_profile_image_filename(name, size, file_extension=file_extension)
> > +        )
> > +        return url + '?v=' + version if version is not None else url
>
> What's the intended behavior if version is None?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/7517/files#r27526512>.
>